tests/unit/clients/workflow_test.py

```diff
@@ -54,3 +77,239 @@ def test_terminate_workflow(workflow_config, workflow_client, boto_client):
         workflowId='workflow_id',
         reason='reason',
     )
+
+
+def test_build_time_filter_dict(oldest_start_date, latest_start_date):
```

Comment:
Since you're already unit testing with different inputs, let's break this up into individual tests.